core/commands/store/path/path.go

```diff
@@ -0,0 +1,210 @@
+			OriginPath = home
+		} else {
+			return fmt.Errorf("can not find the original stored path")
+		}
+
+		if CheckExist(StorePath) == false {
+			err := os.MkdirAll(StorePath, os.ModePerm)
+			if err != nil {
+				return fmt.Errorf("mkdir: %s", err)
+			}
+		} else if CheckDirEmpty(filepath.Join(StorePath, storeDir)) == false {
+			return fmt.Errorf("path is occupied")
+		}
+
+		restartCmd := exec.Command("btfs", "restart")
+		if err := restartCmd.Run(); err != nil {
+			return fmt.Errorf("restart command: %s", err)
+		}
+		return nil
+	},
+}
+
+func init() {
+	GetPropertiesPath()
+	SetEnvVariables()
+}
+
+func WriteProperties() error {
+	if CheckExist(filePath) == false {
+		newFile, err := os.Create(filePath)
+		defer newFile.Close()
+		if err != nil {
+			return err
+		}
+	}
+	data := []byte(StorePath)
+	err := ioutil.WriteFile(filePath, data, 0666)
+	if err == nil {
+		fmt.Printf("Storage location was reset in %v\n", StorePath)
+	}
+	return err
+}
+
+func MoveFolder() error {
+	// make dir does not contain .btfs, but move need to specify .btfs
+	err := os.Rename(filepath.Join(OriginPath, storeDir), filepath.Join(StorePath, storeDir))
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func ReadProperties(filePath string) string {
+	f, err := ioutil.ReadFile(filePath)
+	if err != nil {
+		log.Errorf("Read properties fail: [%v]\n", err)
+	}
+	return string(f)
+}
+
+func CheckDirEmpty(dirname string) bool {
+	dir, _ := ioutil.ReadDir(dirname)
+	if len(dir) == 0 {
+		return true
+	} else {
+		return false
+	}
+}
+
+func Home() (string, error) {
+	user, err := user.Current()
+	if nil == err {
+		return user.HomeDir, nil
+	}
+	if "windows" == runtime.GOOS {
+		return homeWindows()
+	}
+	return homeUnix()
+}
+
+func homeUnix() (string, error) {
+	// First prefer the HOME environmental variable
+	if home := os.Getenv("HOME"); home != "" {
+		return home, nil
+	}
+	// If that fails, try the shell
+	var stdout bytes.Buffer
+	cmd := exec.Command("sh", "-c", "eval echo ~$USER")
+	cmd.Stdout = &stdout
+	if err := cmd.Run(); err != nil {
+		return "", err
+	}
+
+	result := strings.TrimSpace(stdout.String())
+	if result == "" {
+		return "", errors.New("blank output when reading home directory")
+	}
+	return result, nil
+}
+
+func homeWindows() (string, error) {
+	drive := os.Getenv("HOMEDRIVE")
+	path := os.Getenv("HOMEPATH")
+	home := drive + path
+	if drive == "" || path == "" {
+		home = os.Getenv("USERPROFILE")
+	}
+	if home == "" {
+		return "", errors.New("HOMEDRIVE, HOMEPATH, and USERPROFILE are blank")
+	}
+	return home, nil
+}
+
+func GetPropertiesPath() {
+	var propertiesPath string
+	if home, err := Home(); err == nil {
+		if "windows" == runtime.GOOS {
+			propertiesPath = filepath.Join(home, "btfs")
+		} else {
+			propertiesPath = filepath.Join(home, "btfs/bin")
+		}
+	}
+
+	if CheckExist(propertiesPath) == false {
+		err := os.MkdirAll(propertiesPath, os.ModePerm)
+		if err != nil {
+			log.Fatal(err)
```

Comment:
Should not fatal. Even if propertiesPath does not exist, the default ~/.btfs still works. Can log a more detailed error message here.

Reply:
yes, agree, the error will not cause current program to exit.